adam_core/dynamics/stumpff.py

```diff
@@ -0,0 +1,89 @@
+from typing import Tuple
+
+import jax.numpy as jnp
+from jax import config, jit, lax
+
+config.update("jax_enable_x64", True)
+
+STUMPFF_TYPES = Tuple[
+    jnp.float64, jnp.float64, jnp.float64, jnp.float64, jnp.float64, jnp.float64
+]
+
+
+@jit
+def _positive_psi(psi: jnp.float64) -> STUMPFF_TYPES:
+    # Equation 6.9.15 in Danby (1992) [1]
+    sqrt_psi = jnp.sqrt(psi)
+    c0 = jnp.cos(sqrt_psi)
+    c1 = jnp.sin(sqrt_psi) / sqrt_psi
+
+    # Equation 6.9.16 in Danby (1992) [1]
+    # states the recursion relation for higher
+    # order Stumpff functions
+    c2 = (1.0 - c0) / psi
+    c3 = (1.0 - c1) / psi
+    c4 = (1 / 2.0 - c2) / psi
+    c5 = (1 / 6.0 - c3) / psi
+
+    return c0, c1, c2, c3, c4, c5
+
+
+@jit
+def _negative_psi(psi: jnp.float64) -> STUMPFF_TYPES:
+    # Equation 6.9.15 in Danby (1992) [1]
+    sqrt_npsi = jnp.sqrt(-psi)
+    c0 = jnp.cosh(sqrt_npsi)
+    c1 = jnp.sinh(sqrt_npsi) / sqrt_npsi
+
+    # Equation 6.9.16 in Danby (1992) [1]
+    # states the recursion relation for higher
+    # order Stumpff functions
+    c2 = (1.0 - c0) / psi
+    c3 = (1.0 - c1) / psi
+    c4 = (1 / 2.0 - c2) / psi
+    c5 = (1 / 6.0 - c3) / psi
```

Comment:
So many data dependencies here... it makes sense that this was the best in the 60s when Stumpff was working on the Apollo computer, but I bet someone really smart could do better for today's CPU architectures.  (obviously, not a concern for this PR :P)